src/metabase/upload.clj

```diff
@@ -181,15 +187,21 @@
     (contains? (type->ancestors type-b) type-a) type-a
     :else (lowest-common-member (type->ancestors type-a) (type->ancestors type-b))))
 
-(defn- coalesce-types
-  [types-so-far new-types]
-  (->> (map vector types-so-far new-types)
-       (mapv (partial apply lowest-common-ancestor))))
+(defn- map-with-nils
```

Comment:
Introducing this function allows us to get rid of `pad`, which we needed before because `map` finishes iterating when the first collection is exhausted.